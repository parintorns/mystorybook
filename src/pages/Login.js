import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  StyleSheet,
  Image
} from 'react-native'
import {
  View,
  Button,
  Icon,
  Text
} from 'native-base'

export class Login extends Component {
  constructor() {
    super()

    this.handleLoginWithFacebook = this.handleLoginWithFacebook.bind(this)
    this.handleLoginWithGoogle = this.handleLoginWithGoogle.bind(this)
  }

  handleLoginWithFacebook () {
    this.props.navigation.navigate('Register')
  }

  handleLoginWithGoogle() {
    this.props.navigation.navigate('Register')
  }

  render () {
    return (
      <View style={styles.container}>
        <Image style={styles.logo} />
        <Text style={styles.description}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.
        </Text>
        <View style={styles.btnWrapper}>
          <Button
            block
            id='btnFacebook'
            style={[styles.blockButton, styles.facebook]}
            onPress={() => this.handleLoginWithFacebook()}>
            <Icon name='logo-facebook' />
            <Text uppercase={false} style={styles.buttonText}> Sign in with Facebook </Text>
          </Button>
        </View>
        <View style={styles.btnWrapper}>
          <Button
            block
            id='btnGoogle'
            style={[styles.blockButton, styles.google]}
            onPress={() => this.handleLoginWithGoogle()}>
            <Icon name='logo-google' />
            <Text uppercase={false} style={styles.buttonText}> Sign in with Google </Text>
          </Button>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: 20,
    paddingRight: 20
  },
  logo: {
    height: 120,
    width: 120,
    aspectRatio: 1,
    borderRadius: 60,
    backgroundColor: '#ebebeb'
  },
  description: {
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 50,
    color: 'grey'
  },
  blockButton: {
    borderRadius: 40
  },
  buttonText: {
    fontSize: 18,
    color: 'white'
  },
  btnWrapper: {
    marginVertical: 10,
    justifyContent: 'center',
    alignItems: 'center'
  },
  google: {
    justifyContent: 'center',
    backgroundColor: '#dd4b39'
  },
  facebook: {
    justifyContent: 'center',
    backgroundColor: '#3b5998'
  }
})


Login.propTypes = {
  navigation: PropTypes.object.isRequired
}

export default Login
