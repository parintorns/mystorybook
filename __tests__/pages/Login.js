import React from 'react'
import { shallow } from 'enzyme'

import Login from '../../src/pages/Login'

const handleLoginWithFacebook = jest.spyOn(Login.prototype, 'handleLoginWithFacebook')
const handleLoginWithGoogle = jest.spyOn(Login.prototype, 'handleLoginWithGoogle')

const props = {
  navigation: {
    navigate: jest.fn
  }
}

let wrapper

beforeEach(() => {
  wrapper = shallow(
    <Login
      {...props}
    />
  )
})

describe('Render', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Actions', () => {
  it('should go to Facebook login' ,() => {
    const btnFacebook = wrapper.find({ id: 'btnFacebook'})
    expect(btnFacebook).toHaveLength(1)

    btnFacebook.simulate('press')
    expect(handleLoginWithFacebook).toHaveBeenCalledTimes(1)
  })

  it('should go to Google login', () => {
    const btnGoogle = wrapper.find({ id: 'btnGoogle'})
    expect(btnGoogle).toHaveLength(1)

    btnGoogle.simulate('press')
    expect(handleLoginWithGoogle).toHaveBeenCalledTimes(1)
  })
})
