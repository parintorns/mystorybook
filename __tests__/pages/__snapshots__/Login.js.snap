// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    navigation={
      Object {
        "navigate": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          style={
            Object {
              "aspectRatio": 1,
              "backgroundColor": "#ebebeb",
              "borderRadius": 60,
              "height": 120,
              "width": 120,
            }
          }
        />,
        <Styled(Text)
          style={
            Object {
              "alignItems": "center",
              "color": "grey",
              "justifyContent": "center",
              "marginVertical": 50,
            }
          }
          virtual={undefined}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.
        </Styled(Text)>,
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 10,
            }
          }
          virtual={undefined}
        >
          <Styled(Button)
            block={true}
            id="btnFacebook"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 40,
                },
                Object {
                  "backgroundColor": "#3b5998",
                  "justifyContent": "center",
                },
              ]
            }
            virtual={undefined}
          >
            <Styled(Icon)
              name="logo-facebook"
              virtual={undefined}
            />
            <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
              uppercase={false}
              virtual={undefined}
            >
               Sign in with Facebook 
            </Styled(Text)>
          </Styled(Button)>
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 10,
            }
          }
          virtual={undefined}
        >
          <Styled(Button)
            block={true}
            id="btnGoogle"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 40,
                },
                Object {
                  "backgroundColor": "#dd4b39",
                  "justifyContent": "center",
                },
              ]
            }
            virtual={undefined}
          >
            <Styled(Icon)
              name="logo-google"
              virtual={undefined}
            />
            <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
              uppercase={false}
              virtual={undefined}
            >
               Sign in with Google 
            </Styled(Text)>
          </Styled(Button)>
        </Styled(ViewNB)>,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "aspectRatio": 1,
            "backgroundColor": "#ebebeb",
            "borderRadius": 60,
            "height": 120,
            "width": 120,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.",
          "style": Object {
            "alignItems": "center",
            "color": "grey",
            "justifyContent": "center",
            "marginVertical": 50,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Button)
            block={true}
            id="btnFacebook"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 40,
                },
                Object {
                  "backgroundColor": "#3b5998",
                  "justifyContent": "center",
                },
              ]
            }
            virtual={undefined}
          >
            <Styled(Icon)
              name="logo-facebook"
              virtual={undefined}
            />
            <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
              uppercase={false}
              virtual={undefined}
            >
               Sign in with Facebook 
            </Styled(Text)>
          </Styled(Button)>,
          "style": Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 10,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <Styled(Icon)
                name="logo-facebook"
                virtual={undefined}
              />,
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                 Sign in with Facebook 
              </Styled(Text)>,
            ],
            "id": "btnFacebook",
            "onPress": [Function],
            "style": Array [
              Object {
                "borderRadius": 40,
              },
              Object {
                "backgroundColor": "#3b5998",
                "justifyContent": "center",
              },
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "logo-facebook",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Sign in with Facebook ",
                "style": Object {
                  "color": "white",
                  "fontSize": 18,
                },
                "uppercase": false,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": " Sign in with Facebook ",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Button)
            block={true}
            id="btnGoogle"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 40,
                },
                Object {
                  "backgroundColor": "#dd4b39",
                  "justifyContent": "center",
                },
              ]
            }
            virtual={undefined}
          >
            <Styled(Icon)
              name="logo-google"
              virtual={undefined}
            />
            <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
              uppercase={false}
              virtual={undefined}
            >
               Sign in with Google 
            </Styled(Text)>
          </Styled(Button)>,
          "style": Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 10,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <Styled(Icon)
                name="logo-google"
                virtual={undefined}
              />,
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                 Sign in with Google 
              </Styled(Text)>,
            ],
            "id": "btnGoogle",
            "onPress": [Function],
            "style": Array [
              Object {
                "borderRadius": 40,
              },
              Object {
                "backgroundColor": "#dd4b39",
                "justifyContent": "center",
              },
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "logo-google",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Sign in with Google ",
                "style": Object {
                  "color": "white",
                  "fontSize": 18,
                },
                "uppercase": false,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": " Sign in with Google ",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#ebebeb",
                "borderRadius": 60,
                "height": 120,
                "width": 120,
              }
            }
          />,
          <Styled(Text)
            style={
              Object {
                "alignItems": "center",
                "color": "grey",
                "justifyContent": "center",
                "marginVertical": 50,
              }
            }
            virtual={undefined}
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.
          </Styled(Text)>,
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginVertical": 10,
              }
            }
            virtual={undefined}
          >
            <Styled(Button)
              block={true}
              id="btnFacebook"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 40,
                  },
                  Object {
                    "backgroundColor": "#3b5998",
                    "justifyContent": "center",
                  },
                ]
              }
              virtual={undefined}
            >
              <Styled(Icon)
                name="logo-facebook"
                virtual={undefined}
              />
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                 Sign in with Facebook 
              </Styled(Text)>
            </Styled(Button)>
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginVertical": 10,
              }
            }
            virtual={undefined}
          >
            <Styled(Button)
              block={true}
              id="btnGoogle"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 40,
                  },
                  Object {
                    "backgroundColor": "#dd4b39",
                    "justifyContent": "center",
                  },
                ]
              }
              virtual={undefined}
            >
              <Styled(Icon)
                name="logo-google"
                virtual={undefined}
              />
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                 Sign in with Google 
              </Styled(Text)>
            </Styled(Button)>
          </Styled(ViewNB)>,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": 20,
          "paddingRight": 20,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "aspectRatio": 1,
              "backgroundColor": "#ebebeb",
              "borderRadius": 60,
              "height": 120,
              "width": 120,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.",
            "style": Object {
              "alignItems": "center",
              "color": "grey",
              "justifyContent": "center",
              "marginVertical": 50,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc feugiat facilisis augue, at finibus orci molestie sit amet. Nulla efficitur tincidunt justo, vel cursus ligula sodales ut.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Button)
              block={true}
              id="btnFacebook"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 40,
                  },
                  Object {
                    "backgroundColor": "#3b5998",
                    "justifyContent": "center",
                  },
                ]
              }
              virtual={undefined}
            >
              <Styled(Icon)
                name="logo-facebook"
                virtual={undefined}
              />
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                 Sign in with Facebook 
              </Styled(Text)>
            </Styled(Button)>,
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 10,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": true,
              "children": Array [
                <Styled(Icon)
                  name="logo-facebook"
                  virtual={undefined}
                />,
                <Styled(Text)
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                   Sign in with Facebook 
                </Styled(Text)>,
              ],
              "id": "btnFacebook",
              "onPress": [Function],
              "style": Array [
                Object {
                  "borderRadius": 40,
                },
                Object {
                  "backgroundColor": "#3b5998",
                  "justifyContent": "center",
                },
              ],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "logo-facebook",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Sign in with Facebook ",
                  "style": Object {
                    "color": "white",
                    "fontSize": 18,
                  },
                  "uppercase": false,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": " Sign in with Facebook ",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Button)
              block={true}
              id="btnGoogle"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 40,
                  },
                  Object {
                    "backgroundColor": "#dd4b39",
                    "justifyContent": "center",
                  },
                ]
              }
              virtual={undefined}
            >
              <Styled(Icon)
                name="logo-google"
                virtual={undefined}
              />
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                 Sign in with Google 
              </Styled(Text)>
            </Styled(Button)>,
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 10,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": true,
              "children": Array [
                <Styled(Icon)
                  name="logo-google"
                  virtual={undefined}
                />,
                <Styled(Text)
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                   Sign in with Google 
                </Styled(Text)>,
              ],
              "id": "btnGoogle",
              "onPress": [Function],
              "style": Array [
                Object {
                  "borderRadius": 40,
                },
                Object {
                  "backgroundColor": "#dd4b39",
                  "justifyContent": "center",
                },
              ],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "logo-google",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Sign in with Google ",
                  "style": Object {
                    "color": "white",
                    "fontSize": 18,
                  },
                  "uppercase": false,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": " Sign in with Google ",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
